src/main/java/tr/com/srdc/cda2fhir/transform/ResourceTransformerImpl.java

```diff
@@ -918,6 +922,31 @@ public EntryResult tClinicalDocument2Bundle(ClinicalDocument cdaClinicalDocument
 			}
 		}
 
+		for (DocumentationOf docOf : cdaClinicalDocument.getDocumentationOfs()) {
+			if (docOf.getServiceEvent() != null && fhirComp != null) {
+				ServiceEvent cdaServiceEvent = docOf.getServiceEvent();
+				CompositionEventComponent event = new CompositionEventComponent();
+
+				fhirComp.addEvent(event);
+				// documentationOf.serviceEvent.effectiveTime => event.period
+				if (cdaServiceEvent.getEffectiveTime() != null) {
+					Period period = dtt.tIVL_TS2Period(cdaServiceEvent.getEffectiveTime());
+					event.setPeriod(period);
+				}
+
+				// documentationOf.serviceEvent.assignedEntity -> composition.event.detail
+				if (cdaServiceEvent.getPerformers() != null && !cdaServiceEvent.getPerformers().isEmpty()) {
+					for (Performer1 performer : cdaServiceEvent.getPerformers()) {
+						EntityResult entityResult = tAssignedEntity2Practitioner(performer.getAssignedEntity(),
+								bundleInfo);
+						result.updateFrom(entityResult);
+						Reference reference = entityResult.getPractitionerReference();
+						event.addDetail(reference);
+						event.addCode(new CodeableConcept().setText("Primary Care Physician"));
```

Comment:
so I don't know that we want to set this to primary care physician. I think Bill suggested it, but I think it's incorrect. PCPR just means care provisioning, presumably by the doctor referenced herein, but we can't infer that this is actually their primary care physician. The code is optional, so I think we should remove it.

Reply:
this shouldn't be set here either. If you look at where the code is on the gold file, it should only be populated once per each composition. It actually comes in on the gold file as 'burn by fire' and a SNOMED code, but that doesn't map to the HL7 codes so no idea what to do with that information.